src/main/java/com/google/devtools/build/remote/client/LogParserUtils.java

```diff
@@ -68,40 +61,70 @@ private FileInputStream openGrpcFileInputStream() throws ParamException, IOExcep
     return new FileInputStream(filename);
   }
 
+  private static <T extends Message> boolean maybePrintOperation(
+      Operation o, PrintWriter out, Class<T> t) throws IOException {
+    if (o.getResultCase() == ResultCase.ERROR && o.getError().getCode() != Code.OK.value()) {
+      out.printf("Operation contained error: %s\n", o.getError().toString());
+      return true;
+    } else if (o.getResultCase() == ResultCase.RESPONSE && o.getDone()) {
+      out.println("ExecuteResponse extracted:");
+      out.println(o.getResponse().unpack(t).toString());
+      return true;
+    }
+    return false;
+  }
 
   /**
    * Attempt to find and print ExecuteResponse from the details of a log entry for a Watch call. If
    * no Operation could be found in the Watch call responses, or an Operation was found but failed,
    * a failure message is printed.
    */
-  private static void printExecuteResponse(WatchDetails watch, PrintWriter out) throws IOException {
+  private static void printExecuteResponse(V1WatchDetails watch, PrintWriter out)
+      throws IOException {
+    boolean foundWatch = false;
     for (ChangeBatch cb : watch.getResponsesList()) {
       for (Change ch : cb.getChangesList()) {
         if (ch.getState() != State.EXISTS) {
           continue;
         }
         Operation o = ch.getData().unpack(Operation.class);
-        if (o.getResultCase() == ResultCase.ERROR && o.getError().getCode() != Code.OK.value()) {
-          out.printf("Operation contained error: %s\n", o.getError().toString());
-          return;
-        } else if (o.getResultCase() == ResultCase.RESPONSE && o.getDone()) {
-          out.println("ExecuteResponse extracted:");
-          out.println(o.getResponse().unpack(ExecuteResponse.class).toString());
-          return;
-        }
+        foundWatch =
```

Comment:
nit^3: in this case, 

    if (!foundWatch) {
       foundWatch = maybePrintOperation(o, out, com.google.devtools.remoteexecution.v1test.ExecuteResponse.class);
    }

is a bit clearer. Also, allows you to put the return; statement in there, keeping in line with the previous implementation.